### DISCUSSION TITLE
How I change response from response action hook.

### DISCUSSION THREAD

**Discussion by tannontawat** [directus/directus#11165] [Archive]
I have a requirement to check something before send response. can I user `response action hook` to change response. If can change can you give me some example? Or can't change., Can you suggest the solution.

Thanks.

**Comment by rijkvanzanten** [comment]
You can use the `filter` hooks to inject logic to transform payloads before they're being processed: https://docs.directus.io/extensions/hooks/#filter-events. If that's not flexible enough, consider using the `init` hooks to register a custom express middleware that does whatever you need it to 👍🏻

**Reply by aaleksu** [reply]
Is there an example somewhere of what such init hook with custom express middleware could look like?
There aren't that many examples in directus docs.
I need, for example, to adjust response in some cases, but I cannot find any references in docs.
Do you have any suggestions, @rijkvanzanten ?

**Reply by rijkvanzanten** [reply]
The docs examples are the examples that we have available 👍🏻  (cc @phazonoverload)

**Reply by aaleksu** [reply]
Yeah, I understand that. But those examples don't include any response modifying example.
I couldn't figure it out myself.
